tests/artemis-test-support/src/main/java/org/apache/activemq/transport/amqp/client/AmqpClient.java

```diff
@@ -76,6 +77,23 @@ public AmqpConnection connect() throws Exception {
    }
 
    /**
+    * Creates a connection with the broker at the given location, this method initiates a
+    * connect attempt immediately and will fail if the remote peer cannot be reached.
+    *
+    * @param  protocolTracer handler of all the received and sent frames
+    * @throws Exception if an error occurs attempting to connect to the Broker.
+    * @return a new connection object used to interact with the connected peer.
+    */
+   public AmqpConnection connect(final ProtocolTracer protocolTracer) throws Exception {
```

Comment:
@tabish121 what is the way you suggest to intercept frames received from the client?I've searched for something similar and I've found only a state validator,but it is not the same.Maybe I've missed something?